### ISSUE TITLE
Remove deprecated cirdrlist parameter from createLoadBalancerRule API

### ISSUE THREAD

**Issue by onitake** [apache/cloudstack#4036]
##### ISSUE TYPE

 * Improvement Request
 * Documentation Report

##### COMPONENT NAME

~~~
API
~~~

##### CLOUDSTACK VERSION

~~~
4.11
and all later versions
~~~

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Some networking-related APIs, notably `createLoadBalancerRule,` support a parameter called `cidrlist`. This parameter was (presumably) used to automatically manage firewall rules for load balancer rules.

The parameter is deprecated since at least ACS 4.11, leading to an error when trying to use it.

##### STEPS TO REPRODUCE

~~~
create loadbalancerrule publicipid=yyy algorithm=leastconn name=test-lb publicport=9999 privateport=9999 cidrlist=1.2.3.4/32
~~~
Specifying `openfirewall=true` makes no difference.

##### EXPECTED RESULTS

CloudStack should create a firewall rule for the load balancer.

If this is no longer supported, the parameter should be removed.

##### ACTUAL RESULTS

cloudmonkey prints the following error:
~~~
🙈 Error: (HTTP 431, error code 4350) Parameter cidrList is deprecated; if you need to open firewall rule for the specific CIDR, please refer to createFirewallRule command
~~~

##### OTHER COMMENTS

On a related note, I sent a PR that documents how to send multiple CIDRs in cidrlist: #4035 

During preparation of that patch, I identified a few other APIs that use a CIDR_LIST or DEST_CIDR_LIST parameter or return value and may also be affected:
- CreateEgressFirewallRule
- CreateFirewallRule
- CreatePortForwardingRule
- CreateLoadBalancerRule
- CreateIpForwardingRule
- CreateNetworkACL
- UpdateNetworkACLItem
- AuthorizeSecurityGroupEgress
- AuthorizeSecurityGroupIngress
- CreateVpnCustomerGateway
- UpdateVpnCustomerGateway
- FirewallResponse
- FirewallRuleResponse
- LoadBalancerResponse
- NetworkACLItemResponse
- Site2SiteCustomerGatewayResponse
- Site2SiteVpnConnectionResponse

**Comment by DaanHoogland** [comment]
removal of parameters will only happen in a move to version 5.x (if and when). we can stop the error from being returned and silently ignore the parameter or we can add an informational field that state how it was handled but removing a parameter while we continue to support 4.x is a no-go, @onitake.

**Comment by DaanHoogland** [comment]
well, i'd suggest changing it's description to something like "deprecated, will be ignored", ok?

**Comment by onitake** [comment]
I'll open a new PR for that, but can you check #4035 as well, if it makes sense?

**Comment by DaanHoogland** [comment]
added a comment on code and one on travis ;)

**Comment by DaanHoogland** [comment]
@onitake do you consider this closed? #4037 is not merged and has a conflict but @4035 is merged.

**Comment by DaanHoogland** [comment]
@onitake closing this, please reopen if needed.
